### PR TITLE
fix: Persian calendar updates

### PR DESCRIPTION
**Pull request by jason-p-pickering**
Merging in Persian calendar fixes with Dec 12 version of DHIS2 master code.

### CODE REVIEW

**Comment by jason-p-pickering** [comment]
@mortenoh @larshelge I am trying to figure out what we else we need to do to get this merged into core. Could you guys have a look at the fixes we have made to fix the Persian calendar and let us know?

FYI @brajja

**Comment by larshelge** [comment]
This PR has some quality issues like hard-coding (eg. PeriodType.getCalendar().name().equals( "persian" )) and code style problems, and currently has many conflicts. I am closing this for now.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/impl/PersianCalendar.java [hunk 1]
```diff
@@ -171,15 +171,8 @@ public int daysInMonth( int year, int month )
...
-            if ( month < 4 )
-            {
-                newYear = year - 622;
-            }
-            else
-            {
-                newYear = year - 621;
-            }
+            Calendar gregorianCalendar = GregorianCalendar.getInstance();
+            return gregorianCalendar.daysInMonth(year,month);
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/impl/PersianCalendar.java]: Please add some unit tests for this new calendar.

**Comment by jason-p-pickering** [comment]
Can one of the admins verify this patch?

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/impl/PersianCalendar.java [hunk 2]
```diff
@@ -196,9 +189,40 @@ public int weeksInYear( int year )
...
-        DateTime dateTime = toIso( dateTimeUnit )
-            .toJodaDateTime( ISOChronology.getInstance( DateTimeZone.getDefault() ) );
-        return dateTime.getWeekOfWeekyear();
+
+        if (dateTimeUnit.getYear() > START_ISO.getYear()) {
+            Calendar gregorianCalendar = GregorianCalendar.getInstance();
+            return gregorianCalendar.isoWeek(dateTimeUnit);
+        }
+
+        int theFourth = weekday(new DateTimeUnit (dateTimeUnit.getYear(), 1, 4));
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/impl/PersianCalendar.java]: Please follow DHIS 2 code style.

**Pull request closed without merging**